osf/management/commands/backfill_domain_references.py

```diff
@@ -0,0 +1,78 @@
+# -*- coding: utf-8 -*-
+import re
+import logging
+import operator
+from functools import reduce
+
+from django.apps import apps
+from django.db.models import Q
+from django.core.management.base import BaseCommand
+from framework.celery_tasks import app as celery_app
+from osf.models import NotableDomain, DomainReference
+from django.contrib.contenttypes.models import ContentType
+
+logger = logging.getLogger(__name__)
+
+DOMAIN_MATCH_REGEX = re.compile(r'(?P<protocol>\w+://)?(?P<www>www\.)?(?P<domain>[\w-]+\.\w+)(?P<path>/\w*)?')
+DOMAIN_SEARCH_REGEX = r'(http://[^ \'}\[\]\~\(\)\/]+|https://[^ \'}\[\]\~\(\)\/]+)'
+
+
+@celery_app.task()
+def create_notable_domain_with_reference(domain, resource_id, resource_content_type_pk):
+    domain, created = NotableDomain.objects.get_or_create(
+        domain=domain,
+        defaults={'note': NotableDomain.Note.UNKNOWN}
+    )
+    DomainReference.objects.get_or_create(
+        domain=domain,
+        referrer_object_id=resource_id,
+        referrer_content_type_id=resource_content_type_pk
+    )
+
+    if created:
+        logger.info(f'Creating NotableDomain {domain}')
+
+
+def backfill_domain_references(dry_run=False):
+    from osf.models import DraftNode, AbstractNode
+    model_list = {
+        model: model for model in apps.get_models() if
+        hasattr(model, 'SPAM_CHECK_FIELDS') and model not in (DraftNode, AbstractNode)
+    }
+
+    queries = []
+    for model in model_list:
+        query = reduce(
+            operator.or_,
+            (Q(**{f'{field}__regex': DOMAIN_SEARCH_REGEX}) for field in list(model.SPAM_CHECK_FIELDS))
+        )
+        queries.append(model.objects.filter(query))
+
+    for queryset in queries:
+        for resource_data in queryset.values(*list(queryset.query.model.SPAM_CHECK_FIELDS), 'pk'):
```

Comment:
Can we rewrite `_get_spam_content` (which probably means re-writing `get_spam_fields`) to make it usable to do this (and then reuse the task you already wrote) instead of this mess?

Reply:
I'll look into it, but `_get_spam_content` has some conditionals, like whether a redirect link is being used on that resource, that decide what spam content is checkable on an instance-per=instance basis, but I'm pretty sure changing that behavior is trivial it will just need some product approval, or I might do some annotations to decide the condition in the db, but I agree that's a better solution for the long-term.

Reply:
I think it's literally just that addon that's checked , along with the intersection of `saved_fields`. If you make `saved_fields` optional and skip that `intersection` step if it's not provided (`None` distinct from `[]`), then we can use the helper function.

The bigger "issue" will be that `get_spam_fields` and `_get_spam_content` are only defined on the `SpamOverrideMixin`, so will either need to move to the `SpamMixin` with some default casing or else define those functions on other models of interest.